src/Core/Managed/Shared/Extensibility/Implementation/OperationHolder.cs

```diff
@@ -74,17 +78,38 @@ protected virtual void Dispose(bool disposing)
                     if (!this.isDisposed)
                     {
                         var operationTelemetry = this.Telemetry;
+                        bool isActivityEnabled = false;
+#if !NET40
+                        isActivityEnabled = ActivityProxy.TryRun(() =>
+                        {
+                            var currentActivity = Activity.Current;
+                            if (currentActivity == null || operationTelemetry.Id != currentActivity.ParentId ||
+                                operationTelemetry.Context.Operation.Name != Activity.Current.Tags.FirstOrDefault(t => t.Key == "OperationName").Value)
```

Comment:
The expression: ```.Tags.FirstOrDefault(t => t.Key == "OperationName").Value``` appears 4 times in this review. Can it be refactored into a helper method? (Extension method on ```Activity```, perhaps?)